frame/support/procedural/src/pallet/parse/call.rs

```diff
@@ -0,0 +1,237 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::helper;
+use quote::ToTokens;
+use syn::spanned::Spanned;
+
+/// List of additional token to be used for parsing.
+mod keyword {
+	syn::custom_keyword!(DispatchResultWithPostInfo);
+	syn::custom_keyword!(Call);
+	syn::custom_keyword!(OriginFor);
+	syn::custom_keyword!(weight);
+	syn::custom_keyword!(compact);
+	syn::custom_keyword!(T);
+	syn::custom_keyword!(pallet);
+}
+
+/// Definition of dispatchables typically `impl<T: Config> Pallet<T> { ... }`
+pub struct CallDef {
+	/// The where_clause used.
+	pub where_clause: Option<syn::WhereClause>,
+	/// A set of usage of instance, must be check for consistency with trait.
+	pub instances: Vec<helper::InstanceUsage>,
+	/// The index of call item in pallet module.
+	pub index: usize,
+	/// Information on methods (used for expansion).
+	pub methods: Vec<CallVariantDef>,
+	/// The span of the attribute.
+	pub attr_span: proc_macro2::Span,
+}
+
+/// Definition of dispatchable typically: `#[weight...] fn foo(origin .., param1: ...) -> ..`
```

Comment:
Should `#[transaction]` be here?

Reply:
`#[transactional]` doesn't need any specific support from macro, and macro doesn't need to be aware of it as it will simply call foo_transaction, see the dispatchable foo_transactional in frame/support/test/tests/pallet.rs

(though it lacks a test, I'm doing it now)